src/__mocks__/IntersectionObserver.js

```diff
@@ -17,7 +20,7 @@ module.exports.IntersectionObserver = class IntersectionObserver {
   constructor(callback) {
     this.callback = callback
 
-    this.trackedElements = []
+    this.trackedElements = trackedElements
```

Comment:
I'm not a fan of initializing the `IntersectionObserver` with what was potentially previously tracked as it's not quite per spec. But I do like what you've done inside the `LoadableVisibilityComponent` itself where you double observed the tracked element via the `IntersectionObserver` and in `trackedElements`.

Perhaps a different approach would be something like:

```js
class IntersectionObserver {
  constructor(callback) {
    // ...
    this.trackedElements = []
    // ...
  }

  observe(element) {
    this.trackedElements.push(element)
    globallyTrackedElements.push(element)
  }

  unobserve(element) {
    // remove the element from this.trackedElements and globallyTrackedElements
  }
}
```

Then you can assert against `globallyTrackedElements` to ensure that there aren't any leaks 👍